lib/IRGen/GenType.cpp

```diff
@@ -423,6 +423,19 @@ static llvm::Value *computeExtraTagBytes(IRGenFunction &IGF, IRBuilder &Builder,
   return phi;
 }
 
+static Address alignAddress(IRGenFunction &IGF,
+		            Address originAddress,
+			    llvm::Value *shift){
+  auto &Builder = IGF.Builder;
+  auto &IGM = IGF.IGM;
+  auto *alignedAddress = Builder.CreateBitCast(originAddress, IGM.Int8PtrTy).getAddress();
+  if (IGM.Triple.getOS() == llvm::Triple::Linux && !IGM.Triple.isLittleEndian()) {
```

Comment:
Why does the OS have to be Linux?

Reply:
We were not be able to test these changes on the other OS since the team focuses on Linux.